integrationtests/agent/suite_test.go

```diff
@@ -0,0 +1,186 @@
+
+const (
+	DeploymentsNamespace = "fleet-integration-tests"
+)
+
+func TestFleet(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "Fleet Suite")
+}
+
+var _ = BeforeSuite(func() {
+	ctx, cancel = context.WithCancel(context.TODO())
+
+	testEnv = &envtest.Environment{
+		CRDDirectoryPaths:     []string{filepath.Join("..", "..", "charts", "fleet-crd", "templates", "crds.yaml")},
+		ErrorIfCRDPathMissing: true,
+	}
+
+	cfg, err := testEnv.Start()
+	Expect(err).NotTo(HaveOccurred())
+	Expect(cfg).NotTo(BeNil())
+
+	customScheme := scheme.Scheme
+	customScheme.AddKnownTypes(schema.GroupVersion{Group: "fleet.cattle.io", Version: "v1alpha1"}, &v1alpha1.Bundle{}, &v1alpha1.BundleList{})
+	customScheme.AddKnownTypes(schema.GroupVersion{Group: "fleet.cattle.io", Version: "v1alpha1"}, &v1alpha1.BundleDeployment{}, &v1alpha1.BundleDeploymentList{})
+
+	k8sClient, err = client.New(cfg, client.Options{Scheme: customScheme})
+	Expect(err).NotTo(HaveOccurred())
+	Expect(k8sClient).NotTo(BeNil())
+
+	Expect(k8sClient.Create(ctx, &corev1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: DeploymentsNamespace,
+		},
+	})).NotTo(HaveOccurred())
+
+	registerBundleDeploymentController(cfg)
+})
+
+var _ = AfterSuite(func() {
+	cancel()
+	Expect(testEnv.Stop()).ToNot(HaveOccurred())
+})
+
+func registerBundleDeploymentController(cfg *rest.Config) {
+	d, err := discovery.NewDiscoveryClientForConfig(cfg)
+	Expect(err).ToNot(HaveOccurred())
+
+	disc := memory.NewMemCacheClient(d)
+	mapper := restmapper.NewDeferredDiscoveryRESTMapper(disc)
+	dyn, err := dynamic.NewForConfig(cfg)
+	Expect(err).ToNot(HaveOccurred())
+
+	trig := trigger.New(ctx, mapper, dyn)
+	factory := fleet.NewFactoryFromConfigOrDie(cfg)
+	controller = factory.Fleet().V1alpha1().BundleDeployment()
+	restClientGetter := restClientGetter{
+		cfg:        cfg,
+		discovery:  disc,
+		restMapper: mapper,
+	}
+	coreFactory, err := core.NewFactoryFromConfig(cfg)
+	Expect(err).ToNot(HaveOccurred())
+
+	helmDeployer, err := helmdeployer.NewHelm(DeploymentsNamespace, DeploymentsNamespace, DeploymentsNamespace, DeploymentsNamespace, &restClientGetter,
+		coreFactory.Core().V1().ServiceAccount().Cache(), coreFactory.Core().V1().ConfigMap().Cache(), coreFactory.Core().V1().Secret().Cache())
+	Expect(err).ToNot(HaveOccurred())
+
+	wranglerApply, err := apply.NewForConfig(cfg)
+	Expect(err).ToNot(HaveOccurred())
+
+	resources, err := createResources()
+	Expect(err).ToNot(HaveOccurred())
+
+	deployManager := deployer.NewManager(
+		DeploymentsNamespace,
+		DeploymentsNamespace,
+		DeploymentsNamespace,
+		DeploymentsNamespace,
+		controller.Cache(),
+		&lookup{resources: resources},
+		helmDeployer,
+		wranglerApply)
+	bundledeployment.Register(ctx, trig, mapper, dyn, deployManager, controller)
+	err = factory.Start(ctx, 50)
+	Expect(err).ToNot(HaveOccurred())
+}
+
+func createResources() (map[string][]v1alpha1.BundleResource, error) {
+	v1, err := os.ReadFile("assets/deployment-v1.yaml")
```

Comment:
nit: if we use more asset files, it would be nice to keep the asset path in some constant. We have https://github.com/rancher/fleet/blob/master/e2e/testenv/path.go  which could be copied/adapted if we ever have tests in sub directories.

Reply:
https://github.com/rancher/fleet/pull/1183/commits/aa6de314ea37a723c5edc72d891330b63eabf5ba